provider/aws/config.go

```diff
@@ -95,26 +111,53 @@ func newV2Config(awsConfig AWSSessionConfig) (awsv2.Config, error) {
 
 	cfg, err := config.LoadDefaultConfig(context.Background(), defaultOpts...)
 	if err != nil {
-		return awsv2.Config{}, fmt.Errorf("instantiating AWS config: %w", err)
+		return nil, fmt.Errorf("instantiating AWS config: %w", err)
 	}
 
-	if awsConfig.AssumeRole != "" {
-		stsSvc := sts.NewFromConfig(cfg)
-		var assumeRoleOpts []func(*stscredsv2.AssumeRoleOptions)
-		if awsConfig.AssumeRoleExternalID != "" {
-			logrus.Infof("Assuming role %s with external id", awsConfig.AssumeRole)
-			logrus.Debugf("External id: %s", awsConfig.AssumeRoleExternalID)
-			assumeRoleOpts = []func(*stscredsv2.AssumeRoleOptions){
-				func(opts *stscredsv2.AssumeRoleOptions) {
-					opts.ExternalID = &awsConfig.AssumeRoleExternalID
-				},
+	if awsConfig.DomainRolesMap == nil {
+		hostedZonesConfigs = append(hostedZonesConfigs, &AWSZoneConfig{Config: cfg})
+		return hostedZonesConfigs, nil
+	}
+
+	for dom, role := range awsConfig.DomainRolesMap {
```

Comment:
I am fan of short var names but let's use "domain" here

Reply:
Yup sure definitely will change that